esmvaltool/diag_scripts/ocean/diagnostic_maps_quad.py

```diff
@@ -0,0 +1,272 @@
+"""
+Diagnostic Maps quad:
+
+Diagnostic to produce an image showing four maps.
+These plost show latitude vs longitude and the cube value is used as the colour
+scale.
+        model1              model 1 minus model2
+        model2 minus obs    model1 minus obs
+
+Note that this diagnostic assumes that the preprocessors do the bulk of the
+hard work, and that the cube received by this diagnostic (via the settings.yml
+and metadata.yml files) has no time component, a small number of depth layers,
+and a latitude and longitude coordinates.
+
+An approproate preprocessor for a 3D+time field would be:
+preprocessors:
+  prep_map:
+    extract_levels:
+      levels:  [100., ]
+      scheme: linear_extrap
+    time_average:
+
+This tool is part of the ocean diagnostic tools package in the ESMValTool,
+and was based on the plots produced by the Ocean Assess/Marine Assess toolkit.
+
+Author: Lee de Mora (PML)
+        ledm@pml.ac.uk
+        """
+import logging
+import os
+import sys
+import matplotlib
+matplotlib.use('Agg')  # noqa
+import iris
+import cartopy
+
+import matplotlib.pyplot as plt
+import iris.quickplot as qplt
+import numpy as np
+
+import diagnostic_tools as diagtools
+from esmvaltool.diag_scripts.shared import run_diagnostic
+
+# This part sends debug statements to stdout
+logger = logging.getLogger(os.path.basename(__file__))
+logging.getLogger().addHandler(logging.StreamHandler(sys.stdout))
+
+
+def makemapplot(fig, ax, lons, lats, data, title, zrange=[-100, 100],
+                lon0=0., drawCbar=True, cbarlabel='', doLog=False, ):
+
+    if len(lons) == 0:
+        return fig, ax
+    try:
+        if len(lons.compressed()) == 0:
+            return False, False
+    except AttributeError:
+        logger.warning('makemapplot: latitude and longitude not provided.')
+
+    lons = np.array(lons)
```

Comment:
try not to check for lengths of lists but rather use ```if list:``` or ```if not list:```

Reply:
The diagnostic wasn't using this function, so I removed it.